core/server/web/oauth/app.js

```diff
@@ -36,11 +36,8 @@ module.exports = function setupOAuthApp() {
      */
     function googleOAuthMiddleware(clientId, secret) {
         return (req, res, next) => {
-            // TODO: use url config instead of the string /ghost
-
             //Create the callback url to be sent to Google
-            const callbackUrl = new URL(urlUtils.getSiteUrl());
-            callbackUrl.pathname = '/ghost/oauth/google/callback';
+            const callbackUrl = new URL(`${urlUtils.urlFor('admin', true)}oauth/google/callback`);
```

Comment:
Also - it might be worth calling `urlUtils.urlFor('admin', true)` just once and storing it - I can't remember how expensive that call is though - so might not be worth it